warehouse/query-core/src/main/java/datawave/query/iterator/QueryIterator.java

```diff
@@ -937,18 +938,27 @@ public T next() {
                 itrWithContext = Iterators.transform(tupleItr, new EmptyContext<>());
             }
             
-            final IndexOnlyContextCreator contextCreator = new IndexOnlyContextCreator(sourceDeepCopy, getDocumentRange(documentSource), typeMetadataForEval,
-                            compositeMetadata, this, variables, QueryIterator.this);
-            
-            if (exceededOrEvaluationCache != null)
-                contextCreator.addAdditionalEntries(exceededOrEvaluationCache);
-            
-            final Iterator<Tuple3<Key,Document,DatawaveJexlContext>> itrWithDatawaveJexlContext = Iterators.transform(itrWithContext, contextCreator);
-            Iterator<Tuple3<Key,Document,DatawaveJexlContext>> matchedDocuments = statelessFilter(itrWithDatawaveJexlContext, jexlEvaluationFunction);
-            if (log.isTraceEnabled()) {
-                log.trace("arithmetic:" + arithmetic + " range:" + getDocumentRange(documentSource) + ", thread:" + Thread.currentThread());
+            try {
+                IteratorBuildingVisitor iteratorBuildingVisitor = createIteratorBuildingVisitor(getDocumentRange(documentSource), false, this.sortedUIDs);
```

Comment:
We need to double check, but any additional iterators you add in here must adhere to the contract stated at the top of the createDocumentPipeline method: "It is very important that this pipeline can handle resetting the bottom iterator with a new value. This means that hasNext() needs to call the next iterator. The only state that can be maintained is the next value ready after hasNext() has been called. Once next returns the value, the next hasNext() call must call the next iterator again. So for example Iterators.filter() cannot be used as it uses a google commons AbstractIterator that maintains an iterator state (failed, ready, done); use statelessFilter above instead."